### PR TITLE
fixed some clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**
Mostly this is to make the code more readable, but you may like the usage of the Entry API in `src/iota/keymap.rs` and the use of an iterator instead of indexing  in `src/iota/view.rs`.

Cheers!

### CODE REVIEW

File: src/iota/keymap.rs [hunk 1]
```diff
@@ -55,13 +56,14 @@ impl<T: Copy> Trie<T> {
...
                     let keys = &keys[1..];
-
-                    if map.contains_key(&key) {
-                        map.get_mut(&key).unwrap().bind_keys(keys, value);
-                    } else {
-                        let mut node = Box::new(Trie::new());
-                        node.bind_keys(keys, value);
-                        map.insert(key, node);
+                    match map.entry(key) {
+                        Entry::Vacant(v) => {
```
Comment (gchp) [inline src/iota/keymap.rs]: Cool, didn't know about this!

File: src/iota/view.rs [hunk 2]
```diff
@@ -157,17 +157,11 @@ impl View {
...
-
-        for index in 0..width {
-            let mut ch: char = ' ';
-            if index < status_text_len {
-                ch = status_text[index] as char;
-            }
-            self.uibuf.update_cell(index, height, ch, CharColor::Black, CharColor::Blue);
+        for (index, ch) in status_text.iter().enumerate().take(width) {
+            self.uibuf.update_cell(index, height, *ch as char, CharColor::Black, CharColor::Blue);         
         }
```
Comment (gchp) [inline src/iota/view.rs]: This actually changes things slightly...

Basically, I need the status bar to go the full width of the terminal window. In the existing code, I iterate from 0 to the width of the window, and draw the status text for the first portion of that, and for the rest just draw a blank status bar.

With the change you have here, the status bar will only every be as wide as the status text.

I'm not sure if its possible to use iterators and still go the full width of the window even when the status text is much shorter?

**Comment by gchp** [comment]
Thanks for submitting this! I appreciate it.

Didn't know about the entry api, so thanks for showing me that!
I left a comment on the iterator part you added...

Other than that, I'd be happy to merge this!

File: src/iota/view.rs (hunk 2 above)
Comment (gchp) [inline src/iota/view.rs]: If you build and run iota before and after this change, you'll see what I mean about the status bar. It goes full width before, and only as far as the status text after.

**Comment by llogiq** [comment]
I backed out the iterator change.

**Comment by gchp** [comment]
@llogiq sweet, thank you!

**Comment by llogiq** [comment]
You're welcome.

**Pull request merged**